assembly/assembly-main/src/assembly/bin/launcher.sh

```diff
@@ -0,0 +1,126 @@
+#!/bin/sh
+
+# See: https://sipb.mit.edu/doc/safe-shell/
+set -e
+#set +o posix
+
+# Run the finish function if exit signal initiated
+trap exit SIGHUP SIGINT SIGTERM
+
+init_global_variables () {
+
+  # For coloring console output
+  BLUE='\033[1;34m'
+  GREEN='\033[0;32m'
+  NC='\033[0m'
+
+  ### Define various error and usage messages
+  USAGE="
+Usage:
+  launcher [COMMAND]
+     start                              Starts Che server in new console
+     stop                               Stops Che server
+     restart                            Restart Che server
+     update                             Pull latest version of codenvy/che
+
+Docs: http://eclipse.org/che/getting-started.
+"
+
+  DOCKER_HOST_IP=""
+  DEBUG=false
+  CHE_LIB_VOLUME_PATH="/home/user/che/lib"
+  CHE_WS_VOLUME_PATH="/home/user/che/workspaces"
+  CHE_STG_VOLUME_PATH="/home/user/che/storage"
+  DOCKER_SOCKET="/var/run/docker.sock"
+  CHE_SERVER_PORT="8080"
+  CHE_IMAGE_VERSION="nightly"
+}
+
+usage () {
+  echo "${USAGE}"
+}
+
+error_exit () {
+  echo
+  echo "!!!"
+  echo -e "!!! ${1}"
+  echo "!!!"
+  echo
+}
+
+get_docker_host_ip() {
+  DOCKER_HOST_IP=$(docker run --rm --net host \
+                    alpine sh -c \
+                    "ip a show eth0" | \
+                    grep 'inet ' | \
+                    cut -d/ -f1 | \
+                    awk '{ print $2}')
+}
```

Comment:
with docker for mac it returns the correct ip 

but it doesn't work with docker toolbox for mac
> $ docker run --rm --net host    alpine sh -c ifconfig
> docker0   Link encap:Ethernet  HWaddr 02:42:BD:D7:32:B8  
>           inet addr:172.17.0.1  Bcast:0.0.0.0  Mask:255.255.0.0
>           inet6 addr: fe80::42:bdff:fed7:32b8%32597/64 Scope:Link
>           UP BROADCAST MULTICAST  MTU:1500  Metric:1
>           RX packets:49 errors:0 dropped:0 overruns:0 frame:0
>           TX packets:39 errors:0 dropped:0 overruns:0 carrier:0
>           collisions:0 txqueuelen:0 
>           RX bytes:4277 (4.1 KiB)  TX bytes:5483 (5.3 KiB)
> 
> eth0      Link encap:Ethernet  HWaddr 08:00:27:A2:AF:A6  
>           inet addr:10.0.2.15  Bcast:10.0.2.255  Mask:255.255.255.0
>           inet6 addr: fe80::a00:27ff:fea2:afa6%32597/64 Scope:Link
>           UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
>           RX packets:3609 errors:0 dropped:0 overruns:0 frame:0
>           TX packets:1934 errors:0 dropped:0 overruns:0 carrier:0
>           collisions:0 txqueuelen:1000 
>           RX bytes:2580871 (2.4 MiB)  TX bytes:159603 (155.8 KiB)
> 
> eth1      Link encap:Ethernet  HWaddr 08:00:27:70:6A:D8  
>           inet addr:192.168.99.100  Bcast:192.168.99.255  Mask:255.255.255.0
>           inet6 addr: fe80::a00:27ff:fe70:6ad8%32597/64 Scope:Link
>           UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
>           RX packets:1170 errors:0 dropped:0 overruns:0 frame:0
>           TX packets:909 errors:0 dropped:0 overruns:0 carrier:0
>           collisions:0 txqueuelen:1000 
>           RX bytes:182053 (177.7 KiB)  TX bytes:265568 (259.3 KiB)
> 
> lo        Link encap:Local Loopback  
>           inet addr:127.0.0.1  Mask:255.0.0.0
>           inet6 addr: ::1%32597/128 Scope:Host
>           UP LOOPBACK RUNNING  MTU:65536  Metric:1
>           RX packets:0 errors:0 dropped:0 overruns:0 frame:0
>           TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
>           collisions:0 txqueuelen:1 
>           RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)


the working ip is 192.168.99.100 (eth1) in my case